pkg/tfvars/tfvars.go

```diff
@@ -21,80 +25,46 @@ type config struct {
 	IgnitionBootstrap string `json:"ignition_bootstrap,omitempty"`
 	IgnitionMaster    string `json:"ignition_master,omitempty"`
 
-	aws.AWS             `json:",inline"`
-	libvirt.Libvirt     `json:",inline"`
-	openstack.OpenStack `json:",inline"`
+	*aws.AWS             `json:",inline"`
+	*libvirt.Libvirt     `json:",inline"`
+	*openstack.OpenStack `json:",inline"`
 }
 
-// TFVars converts the InstallConfig and Ignition content to
-// terraform.tfvar JSON.
-func TFVars(clusterID string, cfg *types.InstallConfig, osImage, bootstrapIgn, masterIgn string) ([]byte, error) {
+// TFVars generates terraform.tfvar JSON for launching the cluster.
+func TFVars(clusterID string, clusterName string, baseDomain string, machineCIDR *net.IPNet, bridge string, region string, externalNetwork string, trunkSupport string, osImage string, bootstrapIgn string, masterIgn string, masters []clusterapi.Machine) ([]byte, error) {
```

Comment:
This function has quite a few arguments. IMO we should replace some of these with `installconfig.Platform` or atleast document how each one is used.